tasks/restore.yml

```diff
@@ -1,23 +1,47 @@
 ---
+- block:
+    - name: Sync etcd backup file {{ restore_file_name }} from localhost to all etcd nodes
+      synchronize:
+        use_ssh_args: yes
+        src: "{{ restore_file_path }}/{{ restore_file_name }}"
+        dest: "{{ etcd_backup_local_root_dir }}"
+    
+    - name: Fix backup file {{ restore_file_name }} ownership on all etcd nodes
+      file:
+        path: "{{ etcd_backup_local_root_dir }}/{{ restore_file_name }}"
+        owner: etcd
+        group: root
+  become: yes
+  when: restore_file_name is defined and restore_file_path is defined
+
+- name: Set etcd_backup_latest_file fact if using file option
+  set_fact:
+    etcd_backup_latest_file: "{{ etcd_backup_local_root_dir }}/{{ restore_file_name }}"
+  when: restore_file_name is defined and restore_file_path is defined
+
 - name: Stop etcd-member service on all etcd nodes
   become: yes
   service: name=etcd-member state=stopped
 
-- name: Get local backup files list
+- name: Get local default backup files if restore file option is not defined
```

Comment:
tasks in line 20, 34 and 41 can be put in a single block because they deal with local backup files. `when` conditions can be nested.

Reply:
Have combined them into one block:
```
- block:
    - name: Get local default backup files if no user-defined etcd backup file is provided
      become: yes
      find:
        path: "{{ etcd_backup_local_root_dir }}"
        patterns: '*.tgz'
      register: local_backup_files
    
    - name: Abort if no local default etcd backup files are found
      fail:
        msg: "No local default backup files found in {{ etcd_backup_local_root_dir }} and no user-defined etcd backup file is provided"
      when: local_backup_files.files | length == 0
    
    - name: Set etcd_backup_latest_file fact if no user-defined etcd backup file is provided
      set_fact:
        etcd_backup_latest_file: "{{ local_backup_files.files | sort(attribute='mtime', reverse=true) | map(attribute='path') | first }}"

  when: back_file_name is not defined or backup_file_path is not defined
```